src/gluonts/model/deepar/_network.py

```diff
@@ -96,34 +99,22 @@ def __init__(
             len(self.target_shape) <= 1
         ), "Argument `target_shape` should be a tuple with 1 element at most"
 
-        Dropout = {
+        self.Dropout = {
             "ZoneoutCell": ZoneoutCell,
             "RNNZoneoutCell": RNNZoneoutCell,
             "VariationalDropoutCell": VariationalDropoutCell,
             "VariationalZoneoutCell": VariationalZoneoutCell,
         }[self.dropoutcell_type]
 
         with self.name_scope():
-            self.proj_distr_args = distr_output.get_args_proj()
-            self.rnn = mx.gluon.rnn.HybridSequentialRNNCell()
-            for k in range(num_layers):
-                cell = RnnCell(hidden_size=num_cells)
-                cell = mx.gluon.rnn.ResidualCell(cell) if k > 0 else cell
-                # we found that adding dropout to outputs doesn't improve the performance, so we only drop states
-                if "Zoneout" in self.dropoutcell_type:
-                    cell = (
-                        Dropout(cell, zoneout_states=dropout_rate)
-                        if dropout_rate > 0.0
-                        else cell
-                    )
-                elif "Dropout" in self.dropoutcell_type:
-                    cell = (
-                        Dropout(cell, drop_states=dropout_rate)
-                        if dropout_rate > 0.0
-                        else cell
-                    )
-                self.rnn.add(cell)
-            self.rnn.cast(dtype=dtype)
+            self.rnn = self.rnn_block(
```

Comment:
Maybe rename this as `self.encoder` for better readability? Then `self.decoder` is either a reference to or an entirely different block depending on past only features.

Reply:
Thank you. Yes, I struggled with the naming a bit. In the `canonical` DeepAR, `encoder` would sound weird (although I think this would still be fine).